### DISCUSSION TITLE
Sharing rule for poll type "sharing a budget"

### DISCUSSION THREAD

**Discussion by leander-j** [pik-gane/vodle#96] [General]
> Disclaimer: This is a design decision point

I do understand the logic that shares are always given to the highest-rated option to which a specific fraction of participants can agree. But I would argue that the result of this at least looks counter-intuitive and might even not be in the interest of voters.
![Screenshot 2022-04-06 at 11-14-15 vodle – everygroup's consensus](https://user-images.githubusercontent.com/64526813/161941120-517f3bbe-d8d8-4db3-8d2e-303c31e6db18.png)
If two options are agreed to by the same 80% with a similar rating maybe they'd like to distribute their shares on the options. The current implementation implies a "the winner takes it all" logic. I think at the core of this lies the premise that in the end everybody still has exactly one vote/one share. 
Of course, the current logic perfectly follows the basic principle of effective altruism, meaning that if one option is 0.001% more effective (here: more popular) than another one, I will donate all my money to the first option and nothing to the second. The question then is whether democracy/consensus should follow the same principles of maximum effectivity and all-or-nothing

**Comment by leander-j** [comment]
(Thanks for moving!)
I find this discontinuity also problematic in the sense that by altering my vote just by 1% I can move everybody's shares from one option to the other. At least this has a lot of potential for abuse.
In the below example, if I decrease my vote further by 1% (for whatever reasons) the shares for both funds switch in size. And this is not because a threshold of consensus is reached (I am still far within the green bar) but because the average rating (a continuous variable) of LTFF now falls just below that of GWMIF (by less than 0.1)
![Screenshot 2022-04-06 at 14-33-42 vodle – everygroup's consensus](https://user-images.githubusercontent.com/64526813/161975489-8b38829c-98f5-4f9f-8f84-010ff5aa773e.png)

**Reply by mensch72** [reply]
Hi Leander,

I appreciate this discussion! There is a little room for changing some details in the sharing rule. This only applies for polls of type "sharing a budget", not for polls of type "selecting a single winner" (in the latter type of poll, a concentration of winning probability is desirable).

The "sharing a budget" type is known as "portioning" or "mixing" in the social choice literature. We aim at fair portioning rules in which every group of participants can in principle "control" a fair share of the total budget. In the paper **[Aziz et al. (2019)](https://dl.acm.org/doi/abs/10.1145/3328526.3329552)** this property is called "Unanimous Fair Share" (**UFS**). 

The method we currently use in vodle is called "Conditional Utilitarian Rule (**CUT**)" in that paper. Among the methods discussed in that paper, it is the *simplest* one that fulfills the UFS property we desire (see Table 1 in that paper).

Let us consider some alternative methods.

### (a) Equal sharing between approved options: simple but *not* good for us

One possibility would be the following: 

*Each participant owns a share of 1/N, where N is the number of non-abstaining participants (those who approve at least one option). If a participant approves only one option, her share goes to that option. Otherwise, her share is divided equally between the options she approves.*

This is even simpler than CUT since it does not depend on the current ordering of options.

However, we cannot use it for two reasons:
1. It is **not "clone-proof"**. This means if someone clones an option (adds a more or less identical copy of an existing option to the menu), then it can be expected that the participants who will approve this clone are more or less the same as those who approved the original option. This would have the effect that the share of other options will decrease. For example, if some participant approved the original option and some other option X, X got half of that participant's share before adding the clone, but will only get one third of that participant's share after the cloning.
2. It makes the convergence to a **full consensus impossible**. If we expect that every participant approves her favourite option, then all options that are someone's favourite will get a positive share. But this is exactly not what we want! If ten participants have ten different favourites, but there is a good compromise option that all of them like almost as good as their favourite, then we want the whole budget to got to that good compromise option.

It probably also violates other desirable properties listed in the paper.

### (b) "Consistent" sharing between approved options (CSA): more complex but *maybe* good for us

A slight variation of (a) would be following the following idea:

*Each participant owns a share of 1/N, where N is the number of non-abstaining participants (those who approve at least one option). If a participant approves only one option, her share goes to that option. Otherwise, her share is divided between the options she approves **in the same proportions as in the final allocation** of shares that will result from this process.*
 
Notice that this is a *recursive* definition of the final shares, and it is not immediately clear how to actually calculate the shares based on this definition. With some effort though, one can derive the necessary equations and an algorithm that solves them. It might however not have a unique solution, so some additional criterion for resolving this ambiguity might be needed. The algorithm will also be too complicated to be understood by the lay user, so the explanation for users would be the recursive definition given above. 

### (c) The "random priority (RP)" rule from the Aziz paper: apparently similar and also *maybe* good for us

It seems that method (b) (CSA) is closely related to the method called "Random Priority (**RP**)" in the cited paper, which is even more complicated to explain than CSA. Looking at Table 1 from the paper again, we see it has basically the same desirable properties as CUT (except for being harder to compute). One can also show that in contrast to method (a), RP is clone-proof and allows for full consensus. 

I hence conjecture that also CSA will be clone-proof and allow for full consensus. I will try to prove this.

Regarding the issues that you raised, we can note the following about CSA and RP:
1. They do *not* depend on any ordering of options such as an ordering by total approval or the like. So it cannot react drastically to changes in that ordering. This is good.
2. The (recursive) definition of CSA given above and the definition of RP are "smoother" than that of the current method CUT, so it might seem that the final shares depend smoothly on the fractions of participants that approve any possible set of combination of options. This is almost but not completely true. Because the definition is recursive, the final shares may still sometimes react non-smoothly to slight changes of the approvals. But this will occur much more rarely than under the current method CUT.

## Bottom-line

Besides CUT and RP, the literature does not seem to contain any promising alternatives that fulfil the criteria UFS, clone-proofness, and can support full consensus. I was not able to find CSA in the literature so far, but it seems promising as well. So it seems the choice is mainly between CUT, CSA, and RP.

Finally, two short examples of the consequences of using CSA or RP as compared to the current method CUT:
1. Consider 10 participants of which 3 approve only option X, 3 approve only Y, 2 approve X and Z, and 2 approve Y and Z.
  - Then CUT will give X 30%, Y 30%, and Z 40%.
  - CSA would give X 50%, Y 50%, and Z 0%.
  - RP would give X 42%, Y 42%, and Z 16%.
2. Consider 10 participants of which 2 approve only option X, 2 approve only Y, 3 approve X and Z, and 3 approve Y and Z.
  - Then CUT will give X 20%, Y 20%, and Z 60%.
  - CSA would give X 40%, Y 40%, and Z 20%.
  - RP would give X 32%, Y 32%, and Z 36%.

This means CSA and RP would give the "compromise option" Z a smaller share than CUT does. 


*I will consult with the experts on portioning on this. In the meantime, we could collect some other examples of the consequences of CSA and RP vs the current method here if you want.*
